topo/node/srl/srl.go

```diff
@@ -0,0 +1,111 @@
+package srl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	expect "github.com/google/goexpect"
+	topopb "github.com/google/kne/proto/topo"
+	"github.com/google/kne/topo/node"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+func New(pb *topopb.Node) (node.Implementation, error) {
+	cfg := defaults(pb)
+	proto.Merge(cfg, pb)
+	node.FixServices(cfg)
+	return &Node{
+		pb: cfg,
+	}, nil
+}
+
+type Node struct {
+	pb *topopb.Node
+}
+
+func (n *Node) Proto() *topopb.Node {
+	return n.pb
+}
+
+var (
+	spawner = defaultSpawner
+)
+
+func defaultSpawner(command string, timeout time.Duration, opts ...expect.Option) (expect.Expecter, <-chan error, error) {
+	return expect.Spawn(command, timeout, opts...)
+}
+
+func (n *Node) GenerateSelfSigned(ctx context.Context, ni node.Interface) error {
+	return status.Errorf(codes.Unimplemented, "Unimplemented")
+}
+
+func (n *Node) ConfigPush(ctx context.Context, ns string, r io.Reader) error {
+	return status.Errorf(codes.Unimplemented, "Unimplemented")
+}
+
+func (n *Node) CreateNodeResource(_ context.Context, _ node.Interface) error {
+	return status.Errorf(codes.Unimplemented, "Unimplemented")
+}
+
+func (n *Node) DeleteNodeResource(_ context.Context, _ node.Interface) error {
+	return status.Errorf(codes.Unimplemented, "Unimplemented")
+}
+
+func defaults(pb *topopb.Node) *topopb.Node {
+	return &topopb.Node{
+		Constraints: map[string]string{
+			"cpu":    "0.5",
+			"memory": "1Gi",
+		},
+		Services: map[uint32]*topopb.Service{
+			443: {
+				Name:    "ssl",
+				Inside:  443,
+				Outside: node.GetNextPort(),
+			},
+			22: {
+				Name:    "ssh",
+				Inside:  22,
+				Outside: node.GetNextPort(),
+			},
+			57400: {
+				Name:    "gnmi",
+				Inside:  57400,
+				Outside: node.GetNextPort(),
+			},
+		},
+		Labels: map[string]string{
+			"type": topopb.Node_NOKIA_SRL.String(),
+		},
+		Config: &topopb.Config{
+			Image: "srlinux:latest",
+			Command: []string{
+				"/tini",
+			},
+			Args: []string{
```

Comment:
gah where does that come from?

Reply:
this is a combination of the image's entrypoint

```
docker image inspect --format '{{json .ContainerConfig.Entrypoint}}' srlinux:latest
```

and the command that launches SR Linux process specified in [Software Installation Guide](https://infocenter.nokia.com/public/SRLINUX216R1A/topic/com.srlinux.swinstall/html/swinstall-containers.html?cp=9_2_1#lindapra5isvfaxxqso)